### PR TITLE
Replace use of LruMap with a pure LinkedHashMap

### PR DESCRIPTION
**Pull request by Hixie**
...so that we can shed the quiver dependency in flutter's framework.

### CODE REVIEW

**Comment by Hixie** [comment]
@abarth

File: packages/flutter/lib/src/services/image_cache.dart [hunk 1]
```diff
@@ -54,16 +60,39 @@ const int _kDefaultSize = 1000;
...
   ImageResource loadProvider(ImageProvider provider) {
-    return _cache.putIfAbsent(provider, () {
-      return new ImageResource(provider.loadImage());
-    });
+    ImageResource result = _cache[provider];
+    if (result != null) {
+      _cache.remove(provider);
+    } else {
+      if (_cache.length == maximumSize && maximumSize > 0)
+        _cache.remove(_cache.keys.first);
```
Comment (abarth) [inline packages/flutter/lib/src/services/image_cache.dart]: Should we assert that _cache.length never exceeds maximumSize?

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/services/image_cache.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/services/image_cache.dart]: done

**Pull request merged**